### PR TITLE
feat: add tool to migrate 1.x chronograf dashboards and variables

### PR DESCRIPTION
**Pull request by desa**
This PR introduces `chonograf-migrator`, a tool that is used to migrate 1.x chronograf dashboards into their 2.0 equivalents. We attempt to compile any existing influxql queries into influx. If we are unable to, we add the influxql as a comment and note the error that we encountered while compiling.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by desa** [comment]
@ebb-tide @121watts just want to check to make sure the view types I use make sense here

File: pkger/models.go [hunk 1]
```diff
@@ -2545,8 +2545,8 @@ func (q queries) influxDashQueries() []influxdb.DashboardQuery {
 			Text:     qq.Query,
 			EditMode: "advanced",
 		}
-		// TODO: axe thsi buidler configs when issue https://github.com/influxdata/influxdb/issues/15708 is fixed up
-		newQuery.BuilderConfig.Tags = append(newQuery.BuilderConfig.Tags, influxdb.NewBuilderTag("_measurement"))
+		// TODO: axe this buidler configs when issue https://github.com/influxdata/influxdb/issues/15708 is fixed up
```
Comment (ebb-tide) [inline pkger/models.go]:
```suggestion
		// TODO: axe this builder configs when issue https://github.com/influxdata/influxdb/issues/15708 is fixed up
```

Comment (ebb-tide) [reply pkger/models.go]: sorry it just jumped out at me! :)

File: cmd/chronograf-migrator/main.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		if err != nil {
+			return err
+		}
+
+		pkg.Spec.Resources = append(pkg.Spec.Resources, pkger.DashboardToResource(d2, d2.Name))
+
+		for _, v := range vs {
+			if hasVar[v.Name] {
+				// TODO(desa): not sure what we actually want to do here
+				logger.Printf("Found duplicate variables with name %q skipping\n", v.Name)
```
Comment (ebb-tide) [inline cmd/chronograf-migrator/main.go]: reconciliations are hard

File: cmd/chronograf-migrator/dashboard.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+			Values: influxdb.VariableQueryValues{
+				Query:    fmt.Sprintf("// %s", t.Query.Command),
+				Language: "flux",
+			},
+		}
+	case "databases":
+		v.Arguments = &influxdb.VariableArguments{
+			Type: "query",
+			Values: influxdb.VariableQueryValues{
+				Query:    fmt.Sprintf("// SHOW DATABASES %s", t.Query.DB),
```
Comment (jsteenb2) [inline cmd/chronograf-migrator/dashboard.go]: question out of my own ignorance, what are the `//` for?

File: pkger/chronograf.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package pkger
+
+import "github.com/influxdata/influxdb"
+
+// DashboardToResource is an exported call to dashboardToResource used by the
+// chronograf-migrator. It is in a separate file so that it may be maintained distinctly
+// from the internal one.
+func DashboardToResource(d influxdb.Dashboard, name string) Resource {
```
Comment (jsteenb2) [inline pkger/chronograf.go]: can these just be added to the `clone_resource.go` file intead? Feel free to export the functions that this are called here. Don't see any harm in it 👍

File: cmd/chronograf-migrator/dashboard.go (hunk 3 above)
Comment (desa) [reply cmd/chronograf-migrator/dashboard.go]: Just to make the query a comment

File: pkger/chronograf.go (hunk 4 above)
Comment (desa) [reply pkger/chronograf.go]: Yeah, they totally can. I moved them to a separate file so that I didn't keep hitting merge conflicts during rebases.

Comment (jsteenb2) [reply pkger/chronograf.go]: smart mvoe

File: cmd/chronograf-migrator/dashboard.go [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+				Buckets: []string{"bucket"},
+			},
+		}
+		ds = append(ds, d)
+	}
+
+	return ds
+}
+
+type dbrpMapper struct {
```
Comment (jsteenb2) [inline cmd/chronograf-migrator/dashboard.go]: does the nil value not suffice for this type? An odd dep 🤔

File: cmd/chronograf-migrator/main.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/influxdata/influxdb/chronograf/bolt"
+	"github.com/influxdata/influxdb/pkger"
+	"gopkg.in/yaml.v3"
+)
+
+var chronografDBPath string
+var outputFile string
+
+func init() {
+	flag.StringVar(&chronografDBPath, "db", "", "path to the chronograf database")
```
Comment (jsteenb2) [inline cmd/chronograf-migrator/main.go]: might as well axe the `init` funcs and throw the flags in before the parse call 🤷‍♂, food for thought from an anti initer 😬

File: pkger/clone_resource.go [hunk 7]
```diff
@@ -351,7 +350,8 @@ func convertQueries(iQueries []influxdb.DashboardQuery) queries {
 	return out
 }
 
-func dashboardToResource(dash influxdb.Dashboard, name string) Resource {
+// DashboardToResource converts an influxdb.Dashboard to a pkger.Resource.
+func DashboardToResource(dash influxdb.Dashboard, name string) Resource {
```
Comment (jsteenb2) [inline pkger/clone_resource.go]: 🔥

File: pkger/service_test.go [hunk 8]
```diff
@@ -1576,7 +1576,7 @@ func TestService(t *testing.T) {
 					EditMode: "advanced",
 				}
 				// TODO: remove this when issue that forced the builder tag to be here to render in UI.
-				q.BuilderConfig.Tags = append(q.BuilderConfig.Tags, influxdb.NewBuilderTag("_measurement"))
+				q.BuilderConfig.Tags = append(q.BuilderConfig.Tags, influxdb.NewBuilderTag("_measurement", ""))
```
Comment (jsteenb2) [inline pkger/service_test.go]: can we axe this now?

Comment (jsteenb2) [reply pkger/service_test.go]: @hoorayimhelping are we good to go on the builder tags issue we spke of a few days back?

Comment (hoorayimhelping) [reply pkger/service_test.go]: @jsteenb2 i don't think so. we might be looking at it now, but i'm not sure

Comment (jsteenb2) [reply pkger/service_test.go]: no sweat, thanks for the heads up 👍

File: cmd/chronograf-migrator/dashboard.go (hunk 5 above)
Comment (desa) [reply cmd/chronograf-migrator/dashboard.go]: it's a placeholder for when we need to specify how to turn 1.x dbrps into buckets. `nil` would probably work right now.

**Pull request merged**